### PR TITLE
feat(web): unflatten file contents and modernize selection in restore wizard (#323)

### LINKED ISSUE #323: [Enhancement]: Unflatten file contents in restore UI
### Affected area

Backup engine

### Related backup target

Container backups

### What problem are you trying to solve?

Allow the ability to quickly and efficiently restore specific folders/files in the restore UI. The UI currently displays the entire contents of a backup job in one flattened list, which is severely inefficient for exceedingly large folders, slowing down the UI and reducing page responsiveness, producing a poor UX.

Additionally, the page content is confusing and contradictory. It displays "Restore all X files" in the upper right, but within the dropdown itself, none of the files are selected, giving the impression that none of the files would be restored. These 2 separate streams of information should be synchronized, which could be done by selecting all of the content by default.

This does not address the underlying issue, however, which is that the plugin still considers selecting none of the folders as consenting to restoring all folders, for some reason (`Clear to restore everything`). The logic underpinning this entire section should be reconfigured such that it operates along these lines:

1. a check is an affirmative confirmation TO restore
2. the absence of a check is an affirmative confirmation to NOT restore

Image
Image

### What would you like to see?

1. Lazy-loaded file contents in restore UI
2. Unflattened file contents (e.g. branching, click to expand sub-trees)
3. All files / folders selected by default
4. "Clear to restore" text is removed 

### Alternatives considered

_No response_

### Additional context

_No response_

### Confirmation

- [x] I searched existing issues and discussions before opening this enhancement request.
- [x] This is a product improvement request, not a support question.

### PR DESCRIPTION
**Pull request by ruaan-deysel**
## Summary
Resolves #323.

Transforms the file picker in Step 3 of the Restore Wizard from a flat list into a hierarchical, lazy-rendered file tree with intuitive selection semantics:
- **Hierarchical tree structure**: Converts flat tar index entries into nested folder/file trees using `buildFileTree` and `splitPath` in `utils.js`.
- **Lazy DOM rendering**: Only mounted/rendered when a folder is expanded (`FileTree.svelte`), eliminating DOM slowdowns on archives with hundreds or thousands of files.
- **Default all-selected**: When file contents load, all files and folders are selected by default.
- **Inverted selection semantics**: A checked item is an affirmative inclusion; an unchecked item is an affirmative exclusion.
- **Tri-state checkboxes**: Folders display checked (all children selected), indeterminate (subset selected), or unchecked (none selected), with one-click subtree toggle.
- **Search integration**: Typing a search query automatically reveals and expands matching ancestor folders while hiding non-matching items.
- **Batch controls**: Adds "Select all", "Deselect all", "Expand all", and "Collapse all" buttons.
- **Removed confusing messaging**: Eliminates "Clear to restore everything". If 0 files are selected for an opened item, restore is prevented with a clear warning to avoid unintended full restores.
- **Fast-path whole restores**: When all files remain selected (or an item is never opened), whole-archive restore is requested without filter overhead.

## Testing
- Added Vitest unit test suite in `web/src/lib/filetree.test.js` covering tree hierarchy, directory sorting, child counts, and edge cases.
- `npm --prefix web test` (292/292 passed).
- `npm --prefix web run lint` (0 errors, 0 warnings).
- `npm --prefix web run build` (passed).
- `make test` (all Go tests passed).
- `make lint` & `make security-check` (0 issues).

### CODE REVIEW

File: web/src/components/RestoreWizard.svelte [hunk 1]
```diff
@@ -181,45 +183,111 @@
...
         const contents = await api.getRestorePointContents(selectedPoint.jobId, selectedPoint.id, item.name)
-        updateEntry(item, { contents, loading: false })
+        const files = contents?.files || []
+        const tree = buildFileTree(files)
+        // Issue #323: All files / folders selected by default upon loading contents
+        const selected = new SvelteSet()
+        for (const f of files) {
+          if (f?.path) selected.add(f.path)
+        }
+        updateEntry(item, { contents, tree, selected, loading: false })
```
[the remark this answers is not in the document]
Comment (ruaan-deysel) [reply web/src/components/RestoreWizard.svelte]: Addressed in 65c005c and f545b65. `selected` is now seeded from `tree[].descendantLeafPaths`, `totalFiles` and item counts are derived from `descendantLeafCount`, and `selectAllFiles`/`deselectAllFiles` walk the tree's leaf paths.

File: web/src/components/FileTree.svelte [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    if (!query) return true
+    const q = query.toLowerCase()
+    if (node.name.toLowerCase().includes(q) || node.path.toLowerCase().includes(q)) {
+      return true
+    }
+    if (node.isDir && node.children) {
+      return node.children.some(child => matchesSearch(child, query))
+    }
+    return false
+  }
```
[the remark this answers is not in the document]
Comment (ruaan-deysel) [reply web/src/components/FileTree.svelte]: Addressed in d5d9910. Search matches are now precomputed once per query change using an O(N) `$derived.by` traversal that populates a `SvelteMap`, which is reused for node matching and auto-expansion.
